src/api/modules/connect/connect.background.ts

```diff
@@ -24,9 +31,31 @@ const background: BackgroundModuleFunction<void> = async (
 
   if (gateway) isGateway(gateway);
 
-  // Note we are not checking if there are any wallets added anymore, as `requestUserAuthorization` (actually
-  // `createAuthPopup`) do that automatically now and will open the Welcome page and throw a "No wallets" added error if
-  // there are no wallets.
+  // While `requestUserAuthorization` (actually `createAuthPopup`) already does this check below, opening the Welcome
```

Comment:
This check probably needs to go into `disconnect` and `getArweaveConfig`, as those work before the wallet is ready. `getPermissions` too but that one seems to work before the wallet is ready for BE too.

Reply:
https://communitylabs.atlassian.net/browse/ARC-1275